classes/services/PKPSubmissionService.inc.php

```diff
@@ -296,6 +296,50 @@ public function getReviewAssignments($submission) {
 		return $reviewAssignmentDao->getBySubmissionId($submission->getId());
 	}
 
+	/**
+	 * Is this user allowed to view the author details?
+	 *
+	 * - Anyone can view published submission authors
+	 * - Reviewers can only view authors in open reviews
+	 * - Managers and admins can view authors of any submission
+	 * - Subeditors, authors and assistants can only view authors in assigned subs
+	 *
+	 * @param $user User
+	 * @param $submission Submission
+	 * @return boolean
+	 */
+	public function canUserViewAuthor($user, $submission) {
+
+		if ($submission->getDatePublished()) {
+			return true;
```

Comment:
For OJS, the submission should also be assigned to an issue, and that issue should be published.

Reply:
That wouldn't be the case for the continuous publishing model, would it? For issue-based publishing, I don't believe the `datePublished` is set until the issue is actually published.

Reply:
Yes, that would make sense, but we haven't coded in that change yet (unless I've totally missed it!) so for the moment I'd suggest keeping the implementation consistent, and maybe flagging the relevant LOC in the continuous publishing git issue.